### PR TITLE
Fixing links across the docs

### PR DESCRIPTION
**Pull request by leggetter**
This PR:

- Turns on all broken link reporting within Docusaurus 319ad1b60ba164ecc0b278e08c3c2e94789dfd65
- Fixes a bunch of links that were reported

## Problems identified

The main problems with the links are:

- An update from using `documents` to `database` and it seems we just missed a number of links
- Potentially problems following our move from docs.tigrisdata.com to tigrisdata.com/docs however, I do think there were some broken links in place before this. See the next section on lessons learned.

## Lessons learned

A few lessons learned along the way:

- Docusaurus recommend using Markdown links to markdown documents for relative links e.g. `[a link][../path/to/a/file.mdx]`. See https://docusaurus.io/docs/markdown-features/links for the details and a list of benefits
- We have the Docusuarus config `trailingSlash` set to `true`. This means all URLs appear to be directories so relative links need to go up one directory to reference a file that is actually at the same level e.g.
   ```
   ├── http
   │   ├── insert.md
   │   └── query.md
   ```
   For the above, if I want to reference /overview/insert.md within /overview/query.md I can't use `./insert` or `insert` as my href. I have to do `../insert` because I am within `/overview/query/`.
  
   This is another good reason to use links to markdown or MDX files within URLs.

## Link checking output from Docusaurus

Docusaurus outputs link checking feedback (as warnings) in two situations:

1. When you run `npm run start` to develop locally. However, this doesn't seem to tach all problems so it's worth running a build every so often.
2. When you build via `npm run build` for a full build.

Output from `npm run build` below (and some checkboxes I used as I was fixing):

```
[WARNING] Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

It looks like some of the broken links we found appear in many pages of your site.
Maybe those broken links appear on all pages through your site layout?
We recommend that you check your theme configuration for such links (particularly, theme navbar and footer).
Frequent broken links are linking to:
- query/#filter

Exhaustive list of all broken links found:

- On source page path = /docs/cli/authentication/:
   -> linking to reference/ (resolved as: /docs/cli/authentication/reference/) ✅

- On source page path = /docs/database/crud/:
   -> linking to documents/ (resolved as: /docs/database/crud/documents/) ✅
   -> linking to filters/ (resolved as: /docs/database/crud/filters/) ✅
   -> linking to /docs/java/update/ ✅

- On source page path = /docs/database/datamodel/:
   -> linking to ../typescript/database/datamodel/ (resolved as: /docs/database/typescript/database/datamodel/) ✅
   -> linking to ../golang/documents/datamodel/ (resolved as: /docs/database/golang/documents/datamodel/) ✅
   -> linking to ../java/datamodel/declare/ (resolved as: /docs/database/java/datamodel/declare/) ✅

- On source page path = /docs/database/search/:
   -> linking to ../typescript/database/search/#searching-for-documents (resolved as: /docs/database/typescript/database/search/) ✅
   -> linking to filters/ (resolved as: /docs/database/search/filters/) ✅
   -> linking to ../typescript/database/search/ (resolved as: /docs/database/typescript/database/search/) ✅
   -> linking to ../golang/database/search/ (resolved as: /docs/database/golang/database/search/) ✅
   -> linking to ../java/database/search/ (resolved as: /docs/database/java/database/search/) ✅

- On source page path = /docs/database/transaction/:
   -> linking to ../typescript/database/transactions/ (resolved as: /docs/database/typescript/database/transactions/) ✅
   -> linking to ../golang/database/transactions/ (resolved as: /docs/database/golang/database/transactions/) ✅
   -> linking to ../java/database/transactions/ (resolved as: /docs/database/java/database/transactions/) ✅

- On source page path = /docs/golang/:
   -> linking to documents/datamodel/ (resolved as: /docs/golang/documents/datamodel/) ✅
   -> linking to documents/insert/ (resolved as: /docs/golang/documents/insert/) ✅
   -> linking to documents/query/ (resolved as: /docs/golang/documents/query/) ✅
   -> linking to documents/search/ (resolved as: /docs/golang/documents/search/) ✅

- On source page path = /docs/golang/database/search/:
   -> linking to query/#filter (resolved as: /docs/golang/database/search/query/) ✅ resolved in template

- On source page path = /docs/guides/nextjs/add-existing-app/:
   -> linking to /docs/overview/databases/ ✅

- On source page path = /docs/http/datamodel/types/:
   -> linking to /docs/category/client-libraries/ ✅

- On source page path = /docs/http/delete/:
   -> linking to query/#filter (resolved as: /docs/http/delete/query/) ✅ resolved in template

- On source page path = /docs/http/search/:
   -> linking to query/#filter (resolved as: /docs/http/search/query/) ✅ resolved in template

- On source page path = /docs/http/update/:
   -> linking to query/#filter (resolved as: /docs/http/update/query/) ✅ resolved in template

- On source page path = /docs/java/database/:
   -> linking to getting-started/ (resolved as: /docs/java/database/getting-started/) ✅

- On source page path = /docs/java/database/search/:
   -> linking to query/#filter (resolved as: /docs/java/database/search/query/) ✅ resolved in template

- On source page path = /docs/java/getting-started/:
   -> linking to datamodel/declare/ (resolved as: /docs/java/getting-started/datamodel/declare/) ✅

- On source page path = /docs/typescript/database/schema-bootstrap/:
   -> linking to datamodel/#declare-using-decorators (resolved as: /docs/typescript/database/schema-bootstrap/datamodel/) ✅
   -> linking to schema/#modifying-the-schema (resolved as: /docs/typescript/database/schema-bootstrap/schema/) ✅

- On source page path = /docs/typescript/database/search/:
   -> linking to query/#filter (resolved as: /docs/typescript/database/search/query/) ✅ resolved in template
```

### CODE REVIEW

File: docs/database/crud.md [hunk 1]
```diff
@@ -63,18 +63,18 @@ matching zero or more conditions on any field.
...
 from a collection:
 
-- [findOne()](../typescript/database/query#simple-read-query): read a single document
-- [findMany()](../typescript/database/query#filtering-on-multiple-fields): read
+- [findOne()](/docs/typescript/database/query#simple-read-query): read a single document
+- [findMany()](/docs/typescript/database/query#filtering-on-multiple-fields): read
   multiple documents
 
-See the [Filters](filters) section to learn more about the filters supported in
+See the [Filters](filters.md) section to learn more about the filters supported in
```
Comment (ovaistariq) [inline docs/database/crud.md:70]: The extension is not needed

File: docs/database/crud.md [hunk 2]
```diff
@@ -87,17 +87,17 @@ any field.
 Tigris TypeScript SDK provides the following API to update documents in a
 collection:
 
-- [update()](../typescript/database/update#simple-update): update one or more
+- [update()](/docs/typescript/database/update#simple-update): update one or more
   documents that match the filtering condition
 
-See the [Filters](filters) section to learn more about the filters supported in
+See the [Filters](filters.md) section to learn more about the filters supported in
```
Comment (ovaistariq) [inline docs/database/crud.md:93]: Remove the extension

File: docs/database/crud.md [hunk 3]
```diff
@@ -109,13 +109,13 @@ you the flexibility to delete documents matching conditions on any field.
 Tigris TypeScript SDK provides the following API to delete documents from a
 collection:
 
-- [delete()](../typescript/database/delete#simple-delete)
+- [delete()](/docs/typescript/database/delete#simple-delete)
 
-See the [Filters](filters) section to learn more about the filters supported in
+See the [Filters](filters.md) section to learn more about the filters supported in
```
Comment (ovaistariq) [inline docs/database/crud.md:114]: Same as above

File: docs/database/datamodel.mdx [hunk 4]
```diff
@@ -62,6 +62,6 @@ represented.
 Checkout the language-specific reference sections to learn more about
 defining data models in your favorite language.
 
-- [Data modeling in TypeScript](../typescript/database/datamodel)
-- [Data modeling in Go](../golang/documents/datamodel)
-- [Data modeling in Java](../java/datamodel/declare)
+- [Data modeling in TypeScript](../typescript/database/datamodel.mdx)
```
Comment (ovaistariq) [inline docs/database/datamodel.mdx:65]: Remove the extensions here

File: docs/database/search.md [hunk 5]
```diff
@@ -16,14 +16,14 @@ Search operations support the following features:
 Tigris TypeScript SDK provides the following API to search documents
 in a collection:
 
-- [search()](../typescript/database/search#searching-for-documents): search for one
+- [search()](../typescript/database/search.mdx#searching-for-documents): search for one
```
Comment (ovaistariq) [inline docs/database/search.md:19]: Extension needs to be removed

File: docs/database/search.md [hunk 6]
```diff
@@ -16,14 +16,14 @@ Search operations support the following features:
...
   or more documents in a collection
 
-See the [Filters](filters) section to learn more about the filters supported in
+See the [Filters](filters.md) section to learn more about the filters supported in
```
Comment (ovaistariq) [inline docs/database/search.md:22]: Same as above

File: docs/database/search.md [hunk 7]
```diff
@@ -16,14 +16,14 @@ Search operations support the following features:
...
 search operations.
 
 See the language-specific sections for more examples of Search operations:
 
-- [Search in TypeScript](../typescript/database/search)
-- [Search in Go](../golang/database/search)
-- [Search in Java](../java/database/search)
+- [Search in TypeScript](../typescript/database/search.mdx)
```
Comment (ovaistariq) [inline docs/database/search.md:27]: Similar here and the rest of the places

File: docs/templates/_delete-specs.template.mdx [hunk 8]
```diff
@@ -18,5 +18,5 @@
 
 Understanding the syntax:
 
-- The `filter` syntax is similar as documented in the [Query](query#filter) section which allows you to filter and delete
+- The `filter` syntax is similar as documented in the [Query](../query#filter) section which allows you to filter and delete
```
Comment (ovaistariq) [inline docs/templates/_delete-specs.template.mdx:21]: In some places I see you have removed the relative path with absolute path. Do we need to do that here as well?

**Comment by ovaistariq** [comment]
The build will be run as part of the PR, so we probably don't need to run it manually?

**Comment by leggetter** [comment]
@ovaistariq 

> The extension is not needed

The file extension is needed or the link would have to be changed to move up one directory (e.g. `../filters`). 

The **Lessons learned** section of the PR explains the recommendation for using the file extension and also why many of our relative links weren't working.

File: docs/templates/_delete-specs.template.mdx (hunk 8 above)
Comment (leggetter) [reply docs/templates/_delete-specs.template.mdx:21]: I made the changes to use an absolute path before I found the details outlined in the **lessons learned**. If we want to be consistent, I can update those absolute path URLs to be relative.

**Comment by ovaistariq** [comment]
Got it. You can ignore my comments about the extension since you are using file path for links and not URL

> @ovaistariq 
> 
> 
> 
> > The extension is not needed
> 
> 
> 
> The file extension is needed or the link would have to be changed to move up one directory (e.g. `../filters`). 
> 
> 
> 
> The **Lessons learned** section of the PR explains the recommendation for using the file extension and also why many of our relative links weren't working.

File: docs/templates/_delete-specs.template.mdx (hunk 8 above)
Comment (ovaistariq) [reply docs/templates/_delete-specs.template.mdx:21]: It would be good to be consistent and it seems like file paths are better to use.

**Review by ovaistariq — APPROVED** [review]

File: docs/templates/_delete-specs.template.mdx (hunk 8 above)
Comment (leggetter) [reply docs/templates/_delete-specs.template.mdx:21]: I'll do a separate PR with a few updates to just use file paths. I'll also update the README with this as an authoring recommendation.

**Pull request merged**